packages/server/api/src/app/flows/issues/issues-service.ts

```diff
@@ -1,49 +1,113 @@
-import { Issue, IssueStatus, ListIssuesParams, PopulatedIssue } from '@activepieces/ee-shared'
 import { rejectedPromiseHandler } from '@activepieces/server-shared'
-import { ActivepiecesError, ApId, apId, ErrorCode, isNil, SeekPage, spreadIfDefined, TelemetryEventName } from '@activepieces/shared'
+import { ActivepiecesError, apId, ApId, ErrorCode, ExecutioOutputFile, FileType, FlowRun, FlowRunStatus, flowStructureUtil, isNil, Issue, IssueStatus, ListIssuesParams, PopulatedIssue, SeekPage, spreadIfDefined, StepOutputStatus, TelemetryEventName } from '@activepieces/shared'
 import dayjs from 'dayjs'
 import { FastifyBaseLogger } from 'fastify'
+import { LessThan } from 'typeorm'
 import { repoFactory } from '../../core/db/repo-factory'
+import { fileService } from '../../file/file.service'
 import { buildPaginator } from '../../helper/pagination/build-paginator'
 import { paginationHelper } from '../../helper/pagination/pagination-utils'
 import { telemetry } from '../../helper/telemetry.utils'
 import { flowVersionService } from '../flow-version/flow-version.service'
 import { IssueEntity } from './issues-entity'
+
 const repo = repoFactory(IssueEntity)
 
+const extractFailedStepId = async (flowRun: FlowRun, log: FastifyBaseLogger) => {
+    const flowSteps = await fileService(log).getDataOrThrow({
+        projectId: flowRun.projectId, 
+        fileId: flowRun.logsFileId ?? '', 
+        type: FileType.FLOW_RUN_LOG,
+    })
+    const rawData = Buffer.from(flowSteps.data).toString('utf-8')
+    const executionOutput = JSON.parse(rawData) as ExecutioOutputFile
+    const executionState = executionOutput.executionState
+
+    const failedStep = Object.entries(executionState.steps).find(
+        ([_, step]) => step.status === StepOutputStatus.FAILED,
```

Comment:
I think this doesn't work on loops  as they are nested and not directly in top level steps.